lib/driverProviders/driverProvider.ts

```diff
@@ -68,22 +67,20 @@ export abstract class DriverProvider {
    * @public
    * @param webdriver instance
    */
-  quitDriver(driver: WebDriver): wdpromise.Promise<void> {
+  async quitDriver(driver: WebDriver): Promise<void> {
     let driverIndex = this.drivers_.indexOf(driver);
     if (driverIndex >= 0) {
       this.drivers_.splice(driverIndex, 1);
     }
 
-    if (driver.getSession() === undefined) {
-      return wdpromise.when(undefined);
-    } else {
-      return driver.getSession()
-          .then<void>((session_: Session) => {
-            if (session_) {
-              return driver.quit();
-            }
-          })
-          .catch<void>(function(err: Error) {});
+    try {
+      if (driver.getSession() !== undefined) {
+        const session = await driver.getSession();
+        if (session) {
+          return await driver.quit();
+        }
+      }
+    } catch (_) {
```

Comment:
Let's at least add a comment that we're intentionally ignoring any errors when quitting the browser at the end of the test. Also, is that a good idea?

Reply:
Cleaned this up! 

Calling driver.getSession() on a previously quit driver throws errors. So that was removed. If someone called quit in the test (which they probably shouldn't...looking at you interactive tests), then Protractor should try and catch this.